python_modules/dagster/dagster/_core/definitions/op_invocation.py

```diff
@@ -59,31 +121,24 @@ def op_invocation_result(
     context = context or build_op_context()
 
     resource_arg_mapping = {arg.name: arg.name for arg in compute_fn.get_resource_args()}
-    resource_args_from_kwargs = {}
-    for resource_arg in resource_arg_mapping:
-        if resource_arg in kwargs:
-            resource_args_from_kwargs[resource_arg] = kwargs[resource_arg]
-            del kwargs[resource_arg]
 
-    resources_provided_in_multiple_places = (resource_args_from_kwargs) and (context.resource_keys)
+    # Extract any resource inputs from the args and kwargs
```

Comment:
A preamble comment to explain the overall process here might be useful, given the complexity of the codepath.

```
The user is allowed to invoke an op with an arbitrary mix of args and kwargs. The framework must ensure that all of these are correctly categorized as inputs, config, or resource objects and then validated.
 
 * Any resources passes as parameters are also made available to user-defined code in the passed context.
 * Provide high-quality error messages (e.g. if something tried to pass a value to an input typed Nothing)
 * Apply default values appropriately
 * Typechecks of inputs
 
Then it recollects them all into a dictionary and invokes the user-defined function with kwargs only.
```